test/extensions/secondorder/diag_ggn/test_diag_ggn.py

```diff
@@ -89,3 +90,46 @@ def test_diag_ggn_mc(problem):
         backpack_res, backpack_res_mc_avg, atol=MC_ATOL, rtol=MC_RTOL
     )
     problem.tear_down()
+
+
+@pytest.mark.parametrize("problem", PROBLEMS, ids=IDS)
+def test_diag_ggn_mc_batch_light(problem):
+    """Test the MC approximation of individual diagonal of
+    Generalized Gauss-Newton/Fisher with few mc_samples (light version)
+
+    Args:
+        problem (ExtensionsTestProblem): Problem for extension test.
+    """
+    problem.set_up()
+
+    backpack_res = BackpackExtensions(problem).diag_ggn_exact_batch()
+    mc_samples = 1000
+    backpack_res_mc_avg = BackpackExtensions(problem).diag_ggn_mc_batch(mc_samples)
+    # sum the batch, and compare with exact mc
+    check_sizes_and_values(
+        backpack_res, backpack_res_mc_avg, atol=MC_BATCH_ATOL, rtol=MC_LIGHT_RTOL
+    )
+    problem.tear_down()
+
+
+@pytest.mark.montecarlo
+@pytest.mark.parametrize("problem", PROBLEMS, ids=IDS)
+def test_diag_ggn_mc_batch(problem):
+    """Test the MC approximation of Diagonal of Gauss-Newton
+       with more samples (slow version)
+
+    Args:
+        problem (ExtensionsTestProblem): Problem for extension test.
+    """
+    problem.set_up()
+
+    backpack_res = BackpackExtensions(problem).diag_ggn_exact_batch()
+    # NOTE May crash for large networks because of large number of samples.
+    # If necessary, resolve by chunking samples into smaller batches + averaging
+    mc_samples = 100000
```

Comment:
I have experienced this now when executing the tests on GPU, and [fixed](https://github.com/f-dangel/backpack/commit/9ad9ee8e89f0a9445f4acca0f34371ca0e1ff383) it for `DiagGGNMC`. The chunking is implemented [here](https://github.com/f-dangel/backpack/blob/branching/test/extensions/secondorder/diag_ggn/test_diag_ggn.py#L92-L94). 

Please also implement the chunking for `DiagGGNMCBatch`, such that these tests can be executed on a GPU.